test/contrib/canfdsocket_native.uts

```diff
@@ -0,0 +1,147 @@
+% Regression tests for nativecanfdsocket
+~ python3_only not_pypy vcan_socket needs_root linux
+
+# More information at http://www.secdev.org/projects/UTscapy/
+
+
+############
+############
++ Configuration of CAN virtual sockets
+~ conf
+
+= Load module
+load_layer("can", globals_dict=globals())
+conf.contribs['CANSocket'] = {'use-python-can': False}
+from scapy.contrib.cansocket_native import *
+conf.contribs['CAN'] = {'swap-bytes': False, 'remove-padding': True}
+
+
+= Setup string for vcan
+bashCommand = "/bin/bash -c 'sudo modprobe vcan; sudo ip link add name vcan0 type vcan; sudo ip link set dev vcan0 up'"
+
+= Load os
+import os
+import threading
+from time import sleep
+from subprocess import call
+
+= Setup vcan0
+assert 0 == os.system(bashCommand)
+
++ Basic Packet Tests()
+= CAN FD Packet init
+canfdframe = CANFD(identifier=0x7ff,length=8,data=b'\x01\x02\x03\x04\x05\x06\x07\x08\xaa')
+assert bytes(canfdframe) == b'\x00\x00\x07\xff\x08\x04\x00\x00\x01\x02\x03\x04\x05\x06\x07\x08\xaa'
+
++ Basic Socket Tests()
+= CAN FD Socket Init
+sock1 = CANSocket(channel="vcan0", fd=True)
+
+= CAN Socket send recv small packet without remove padding
+
+conf.contribs['CAN'] = {'swap-bytes': False, 'remove-padding': False}
+
+sock2 = CANSocket(channel="vcan0", fd=True)
+sock2.send(CANFD(identifier=0x7ff,length=9,data=b'\x01\x02\x03\x04\x05\x06\x07\x08\xaa'))
+sock2.send(CAN(identifier=0x7ff,length=8,data=b'\x01\x02\x03\x04\x05\x06\x07\x08'))
+sock2.close()
+
+rx = sock1.recv()
+assert rx == CANFD(identifier=0x7ff,length=9,data=b'\x01\x02\x03\x04\x05\x06\x07\x08\xaa') / Padding(b"\x00" * (64 - 9))
+rx = sock1.recv()
+assert rx == CAN(identifier=0x7ff,length=8,data=b'\x01\x02\x03\x04\x05\x06\x07\x08')
```

Comment:
@polybassa sorry to bother you but this test seems to fail when the vcan kernel module is available on a machine with
```sh
    >>> rx = sock1.recv()
    >>> assert rx == CANFD(identifier=0x7ff,length=9,data=b'\x01\x02\x03\x04\x05\x06\x07\x08\xaa') / Padding(b"\x00" * (64 - 9))
    >>> rx = sock1.recv()
    >>> assert rx == CAN(identifier=0x7ff,length=8,data=b'\x01\x02\x03\x04\x05\x06\x07\x08')
    Traceback (most recent call last):
      File "<input>", line 2, in <module>
    AssertionError
```

I fixed it locally by sending `CANFD` instead of `CAN` and making sure the padding is kept (because `'remove-padding'` is `False`) but it could be that I'm missing something:
```diff
diff --git a/test/contrib/canfdsocket_native.uts b/test/contrib/canfdsocket_native.uts
index 1f0ca94d..32f90aea 100644
--- a/test/contrib/canfdsocket_native.uts
+++ b/test/contrib/canfdsocket_native.uts
@@ -43,13 +43,13 @@ conf.contribs['CAN'] = {'swap-bytes': False, 'remove-padding': False}

 sock2 = CANSocket(channel="vcan0", fd=True)
 sock2.send(CANFD(identifier=0x7ff,length=9,data=b'\x01\x02\x03\x04\x05\x06\x07\x08\xaa'))
-sock2.send(CAN(identifier=0x7ff,length=8,data=b'\x01\x02\x03\x04\x05\x06\x07\x08'))
+sock2.send(CANFD(identifier=0x7ff,length=8,data=b'\x01\x02\x03\x04\x05\x06\x07\x08'))
 sock2.close()

 rx = sock1.recv()
 assert rx == CANFD(identifier=0x7ff,length=9,data=b'\x01\x02\x03\x04\x05\x06\x07\x08\xaa') / Padding(b"\x00" * (64 - 9))
 rx = sock1.recv()
-assert rx == CAN(identifier=0x7ff,length=8,data=b'\x01\x02\x03\x04\x05\x06\x07\x08')
+assert rx == CANFD(identifier=0x7ff,length=8,data=b'\x01\x02\x03\x04\x05\x06\x07\x08') / Padding(b"\x00" * (64 - 8))


 = CAN Socket send recv

```
If this patch makes sense I can send a PR.

Reply:
I seem to have forgotten to mention what `rx` looks like when the test fails:
```sh
failed 418B60CD 000.01s CAN Socket send recv small packet without remove padding
Test 'CAN Socket send recv small packet without remove padding' failed. exit() to stop, Ctrl-D to leave this interpreter and continue with the current test campaign
>>> rx
<CANFD  flags= identifier=0x7ff length=8 fd_flags=fd_frame reserved=0 data='\x01\x02\x03\x04\x05\x06\x07\x08' |<Padding  load='\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00' |>>
```

Reply:
Thanks for your message.

Unfortunately, I've discovered different ways, how the kernel sets "fd_flags", depending on it's version. This means, there is an inconsistency between machines, if a CAN or a CANFD packet is received. 

I don't see a proper solution for now, other than making the unit tests so generic, that both possible packets are accepted.

Reply:
Actually, I've fixed it in #3928